ui/src/test/java/edu/wpi/grip/ui/codegeneration/BlurGenerator.java

```diff
@@ -52,18 +62,36 @@ public void setUp(){
         injector.injectMembers(gen);
     }
 	@Test
-	public void generateAndRun() {
-		Step step = gen.addStep(new OperationMetaData(BlurOperation.DESCRIPTION, () -> new BlurOperation(isf,osf)));	
-		eventBus.post(new SourceAddedEvent(imgfac.create(new File("src/test/resources/edu/wpi/grip/images/gompei.jpeg"))));
+	public void generateAndRun() throws IOException{
+		Step step = gen.addStep(new OperationMetaData(BlurOperation.DESCRIPTION, () -> new BlurOperation(isf,osf)));
+		ImageFileSource img = imgfac.create(Files.gompeiJpegFile.file);
+		img.initialize();
+		eventBus.post(new SourceAddedEvent(img));
 		OutputSocket imgOut = pipeline.getSources().get(0).getOutputSockets().get(0);
+		
 		for(InputSocket sock : step.getInputSockets()){
 			if(sock.getSocketHint().isCompatibleWith(imgOut.getSocketHint())){
 				gen.connect(imgOut, sock);
 			}
+			else if(sock.getSocketHint().isCompatibleWith(Outputs.createNumberSocketHint("Number", new Double(5.1)))){
+				sock.setValue(new Double(10.0));
+			}
 		}
 		assertEquals("Connection was not added",1,pipeline.getConnections().size());
 		gen.export("Blur.java");
 		PipelineInterfacer pip = new PipelineInterfacer("Blur.java");
+		ManualPipelineRunner runner = new ManualPipelineRunner(eventBus, pipeline);
+		runner.runPipeline();
+		Optional out = step.getOutputSockets().get(0).getValue();
+		assertTrue("Pipeline did not process", out.isPresent());
+		assertFalse("Pipeline has null output", out.get()==null);
+		assertFalse("Pipeline output is empty", ((org.bytedeco.javacpp.opencv_core.Mat)out.get()).empty());
+		pip.setMatSource(0, Files.gompeiJpegFile.file);
+		pip.process();
+		Mat genMat = (Mat) pip.getOutput(0);
```

Comment:
Seeing this makes me think the API is pretty vague -- what is output 0? How are users supposed to know which output is which? It would be much better to have these named after the step, something like `pip.getBlurOutput()` would be _so_ much more descriptive. If there is more than one occurrence of an operation, they can be `getFirstBlurOutput`, `getSecondBlurOutput`, etc.